CMakeLists.txt

```diff
@@ -4,20 +4,18 @@ project(Seasocks VERSION 1.4.3)
 if(WIN32)
     message(FATAL_ERROR "${PROJECT_NAME} does not support Windows")
 endif()
+if(APPLE)
+    message(FATAL_ERROR "${PROJECT_NAME} does not support Apple platforms")
+endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-option(SEASOCKS_SHARED "Build shared Seasocks library" ON)
-option(SEASOCKS_STATIC "Build static Seasocks library" ON)
+option(SEASOCKS_SHARED "Build seasocks as a shared library" "${BUILD_SHARED_LIBS}}")
```

Comment:
@mattgodbolt 
```suggestion
option(SEASOCKS_SHARED "Build seasocks as a shared library" "${BUILD_SHARED_LIBS}")
```

Reply:
@mattgodbolt 
Can you check this post-merge review?
Because there are still some bugs present in the cmake script.